homeassistant/components/linear_garage_door/coordinator.py

```diff
@@ -0,0 +1,81 @@
+"""DataUpdateCoordinator for Linear."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from linear_garage_door import Linear
+from linear_garage_door.errors import InvalidLoginError, ResponseError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LinearUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """DataUpdateCoordinator for Linear."""
+
+    _email: str
+    _password: str
+    _device_id: str
+    _site_id: str
+    _devices: list[dict[str, list[str] | str]] | None
+    _linear: Linear
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize DataUpdateCoordinator for Linear."""
+        self._email = entry.data["email"]
+        self._password = entry.data["password"]
+        self._device_id = entry.data["device_id"]
+        self._site_id = entry.data["site_id"]
+        self._devices = None
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="Linear Garage Door",
+            update_interval=timedelta(seconds=60),
+        )
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Get the data for Linear."""
+
+        linear = Linear()
+
+        try:
+            await linear.login(
```

Comment:
Do you need to log in on each update?

Reply:
Yes. This initiates the connection. This is what the real app does as well

Reply:
Is it possible to keep the connection open?

Reply:
Unfortunately not. After a door opens or closes, it completely stops sending events, and stops accepting all commands